python/cudf/cudf/core/index.py

```diff
@@ -1411,24 +1466,186 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class NumericIndex(GenericIndex):
+    """Immutable, ordered and sliceable sequence of labels.
+    The basic object storing row labels for all cuDF objects.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype,
+            but not used.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+    """
+
+    def __new__(cls, data=None, dtype=None, copy=False, name=None):
+
+        out = Frame.__new__(cls)
+        dtype = _index_to_dtype[cls]
+        if copy:
+            data = column.as_column(data, dtype=dtype).copy()
+
+        kwargs = _setdefault_name(data, name=name)
+
+        data = column.as_column(data, dtype=dtype)
+
+        out._initialize(data, **kwargs)
+
+        return out
+
+
+@copy_docstring(NumericIndex)
```

Comment:
If there's no class / __init__ docstrings defined for a class they're inherited from the base class for `help(class)`, IPython `?` built-in, as well as Sphinx. Is there a reason we need to explicitly copy the docstring here?

Reply:
Yeah I saw this too. Although they aren't explicitly available as the `__doc__` attribute, I guess in practice it doesn't matter.

Reply:
I'm OK with just omitting the `copy_docstring` here and relying on this behaviour

Reply:
I don't think it inherits, for example
```
In [1]: class A(object): 
   ...:     """Class A Doc""" 
   ...:                                                                                                                                                                                                                          

In [2]: class A(object): 
   ...:     def __init__(): 
   ...:         pass 
   ...:          
   ...:                                                                                                                                                                                                                          

In [3]: class B(A): 
   ...:     def __init__(): 
   ...:         pass 
   ...:          
   ...:                                                                                                                                                                                                                          

In [4]: class A(object): 
   ...:     """Class A Doc""" 
   ...:                                                                                                                                                                                                                          

In [5]: class B(A): 
   ...:     def __init__(): 
   ...:         pass 
   ...:          
   ...:                                                                                                                                                                                                                          

In [6]: B.__doc__                                                                                                                                                                                                                

In [7]: A.__doc__                                                                                                                                                                                                                
Out[7]: 'Class A Doc'

```

Reply:
It's not available explicitly as the `__doc__` attribute, but all of the uses of that attribute know to go to the parent class if `__doc__` is None.

Reply:
But as you said with `?` it works 
```
In [8]: B?                                                                                                                                                                                                                       
Init signature: B()
Docstring:      Class A Doc
Type:           type
Subclasses:   
```

Reply:
In that I can remove these